### ISSUE TITLE
使用transformers新Linear Schedule

### ISSUE THREAD

**Issue by flydsc** [yangjianxin1/GPT2-chitchat#26]
在之前的[GPT2-Chinese](https://github.com/Morizeyao/GPT2-Chinese)项目中transformer版本定在2.1.1中，在本项目中是否可以考虑升级？
其实应该就是263行的：
`scheduler = transformers.WarmupLinearSchedule(optimizer, warmup_steps=args.warmup_steps, t_total=total_steps)`
改为
`scheduler = transformers.get_linear_schedule_with_warmup(optimizer, num_warmup_steps=args.warmup_steps, num_training_steps=total_steps)`

改一下几个参数的传入名称即可。

详细文档可以参考如下

1. [2.1.1使用的线性schedule](https://huggingface.co/transformers/v2.1.1/main_classes/optimizer_schedules.html#transformers.WarmupLinearSchedule)

`class WarmupLinearSchedule(LambdaLR):
    """ Linear warmup and then linear decay.
        Linearly increases learning rate from 0 to 1 over `warmup_steps` training steps.
        Linearly decreases learning rate from 1. to 0. over remaining `t_total - warmup_steps` steps.
    """
    def __init__(self, optimizer, warmup_steps, t_total, last_epoch=-1):
        self.warmup_steps = warmup_steps
        self.t_total = t_total
        super(WarmupLinearSchedule, self).__init__(optimizer, self.lr_lambda, last_epoch=last_epoch)

    def lr_lambda(self, step):
        if step < self.warmup_steps:
            return float(step) / float(max(1, self.warmup_steps))
        return max(0.0, float(self.t_total - step) / float(max(1.0, self.t_total - self.warmup_steps)))`

2. [最新版的线性schedule](https://huggingface.co/transformers/main_classes/optimizer_schedules.html#transformers.get_linear_schedule_with_warmup)
```
def get_linear_schedule_with_warmup(optimizer, num_warmup_steps, num_training_steps, last_epoch=-1):
    """ Create a schedule with a learning rate that decreases linearly after
    linearly increasing during a warmup period.
    """

    def lr_lambda(current_step):
        if current_step < num_warmup_steps:
            return float(current_step) / float(max(1, num_warmup_steps))
        return max(
            0.0, float(num_training_steps - current_step) / float(max(1, num_training_steps - num_warmup_steps))
        )

    return LambdaLR(optimizer, lr_lambda, last_epoch)
```

**Comment by ly19901990** [comment]
我是改用的get_linear_schedule_with_warmup，WarmupLinearSchedul因为transformer版本低会报错。

**Comment by yangjianxin1** [comment]
谢谢，在新的代码中更新了
